File: Selenium/Project/Activity7.py
# Import necessary packages
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Start the driver using with statement
with webdriver.Firefox() as driver:
    # Open the page
    driver.get("http://alchemy.hguy.co/crm")
    
    # Find username and password fields
    username_field = driver.find_element(By.ID, "user_name")
    password_field = driver.find_element(By.ID, "username_password")
    login_button = driver.find_element(By.ID, "bigbutton")
    
    # Enter credentials
    username_field.send_keys("admin")
    password_field.send_keys("pa$$w0rd")
    
    # Click the submit button
    login_button.click()
    
    # Find the "Sales" menu item
    sales_menu = driver.find_element(By.ID, "grouptab_0")
    
    # Hover over the "Sales" menu item to reveal the "Leads" option
    actions = ActionChains(driver)
    actions.move_to_element(sales_menu).perform()
    
    # Wait for the "Leads" option to be visible and click it
    wait1 = WebDriverWait(driver, 30)
    lead_option = wait1.until(EC.visibility_of_element_located((By.XPATH, "//*[@id='moduleTab_9_Leads']")))
    lead_option.click()
    
    # Wait for the "Additional Details" button to be visible and click it
    wait2 = WebDriverWait(driver, 10)
    additional_info = wait2.until(EC.visibility_of_element_located((By.CSS_SELECTOR, "[title='Additional Details']")))
    additional_info.click()
    
    # Find the phone number element and print its text
    phone_number = driver.find_element(By.CSS_SELECTOR, ".phone")
    print("The phone number displayed on the leads pop up is:", phone_number.text)

File: Selenium/Project/Activity7.java
package FST_Selenium_Project;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity7 {
	WebDriver driver;
	
	@BeforeClass
	public void setUp() {
		driver = new FirefoxDriver();
		driver.get("http://alchemy.hguy.co/crm");
		WebElement userName = driver.findElement(By.id("user_name"));
		WebElement password = driver.findElement(By.id("username_password"));
		WebElement login = driver.findElement(By.id("bigbutton"));
		
		userName.sendKeys("admin");
		password.sendKeys("pa$$w0rd");
		login.click();
	}
	 @Test
	 public void salesMenu() throws InterruptedException {
		 WebElement salesMenu = driver.findElement(By.id("grouptab_0"));
		 Actions actions = new Actions(driver);
	        actions.moveToElement(salesMenu).perform();
	        WebDriverWait wait1 = new WebDriverWait(driver, Duration.ofSeconds(30));
		 WebElement  leadOption=  wait1.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[@id=\"moduleTab_9_Leads\"]")));
		   leadOption.click();
		   WebDriverWait wait2 = new WebDriverWait(driver, Duration.ofSeconds(10));
		   System.out.println("found button");
	        WebElement additionalInfo = wait2.until(ExpectedConditions.visibilityOfElementLocated(
	        		By.xpath("//*[@id=\"adspan_77c0ebaa-448d-740d-a00f-605ae70266e9\"]/span")));
	        additionalInfo.click();
	        WebDriverWait wait3 = new WebDriverWait(driver, Duration.ofSeconds(100));
	        WebElement phoneNumber = driver.findElement(By.cssSelector(".phone"));
	        System.out.println("The phone number displayed on the leads pop up is: "+phoneNumber.getText());
			
	 }
	 
	
	 
	@AfterClass
	public void tearDown() {
		driver.quit();
	}
}
